### ISSUE TITLE
Converting Oracle schema in oracle 19c to SqlServer, raise an Exception: System.NullReferenceException: Object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by moh-hassan** [martinjw/dbschemareader#124]
I  try to convert HR oracle schema of Oracle database v19c to SqlServer using the nuget packages: 
 `DatabaseSchemaReader v2.7.9`  and ODAC `Oracle.ManagedDataAccess.Client` v19.12.0  in a console net472, with the next code:

```csharp
string connectionString="Data Source=pdb1;User Id=HR;Password=hr;";
string owner="HR";
        
 string providername = "Oracle.ManagedDataAccess.Client";
 var dbReader = new DatabaseReader(connectionString, providername);
dbReader.Owner = owner;
var schema = dbReader.ReadAll();
var gen = new DdlGeneratorFactory(SqlType.SqlServer).AllTablesGenerator(schema);
var text = gen.Write(); // <<this raise the error
```
I get error Exception:

```
Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at DatabaseSchemaReader.SqlGen.SqlServer.ConstraintWriter.<>c.<WritePrimaryKey>b__3_0(DatabaseIndex x)
   at System.Collections.Generic.List`1.Find(Predicate`1 match)
   at DatabaseSchemaReader.SqlGen.SqlServer.ConstraintWriter.WritePrimaryKey(DatabaseConstraint constraint)
   at DatabaseSchemaReader.SqlGen.SqlServer.TableGenerator.ConstraintWriter()
   at DatabaseSchemaReader.SqlGen.TableGeneratorBase.Write()
   at DatabaseSchemaReader.SqlGen.SqlServer.TableGenerator.Write()
   at DatabaseSchemaReader.SqlGen.TablesGeneratorBase.Write()
   at OraclePs.SchemaMigration2.OraSchema2SqlServer(String connectionString, String owner) in 
.....
```

When debugging using the source, the next  errror is fired
```
System.NullReferenceException: 'Object reference not set to an instance of an object.'
DatabaseSchemaReader.DataSchema.DatabaseIndex.IndexType.get returned null.
and point to line 36:
var pkIndex = Table.Indexes.Find(x => x.IndexType.IndexOf("PRIMARY", StringComparison.OrdinalIgnoreCase) != -1);     
                                        ^^^^^^^
```

It seems that IndexType is not initialized and cause this error.

**Problem Resolution**
using Null-conditional operators ?. in the class `ConstraintWriter.cs`, line 37 solve the problem
```
var pkIndex = Table.Indexes.Find(x => x.IndexType?.IndexOf("PRIMARY", StringComparison.OrdinalIgnoreCase) != -1);
```

**Comment by moh-hassan** [comment]
Thanks. It's working fine

**Comment by martinjw** [comment]
Nuget published v 2.7.10
